vertical-pod-autoscaler/pkg/apis/autoscaling.k8s.io/v1/types.go

```diff
@@ -147,6 +147,9 @@ type ContainerResourcePolicy struct {
 	// for the container. The default is no maximum.
 	// +optional
 	MaxAllowed v1.ResourceList `json:"maxAllowed,omitempty" protobuf:"bytes,4,rep,name=maxAllowed,casttype=ResourceList,castkey=ResourceName"`
+	// Whether autoscaler limit scaling is enabled for the container. The default is "Auto".
```

Comment:
If the ContainerScalingMode = "Off", the ContainerLimitScalingMode will be ignored.

We need to make it explicit here in the comments that Mode = "Auto" is a prerequisite for limit scaling and ideally we could reject objects that have:

Mode = "Off"
LimitMode = "Auto"

Validation is performed in VPA admission controller.

Reply:
Validation code: https://github.com/kubernetes/autoscaler/blob/master/vertical-pod-autoscaler/pkg/admission-controller/resource/vpa/handler.go#L109